src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -5615,6 +5615,7 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
       val statsEnabled = StatisticsStatics.areSomeHotStatsEnabled() && statistics.areHotStatsLocallyEnabled
       val startByType = if (statsEnabled) statistics.pushTimer(byTypeStack, byTypeNanos(tree.getClass)) else null
       if (statsEnabled) statistics.incCounter(visitsByType, tree.getClass)
+      if (Thread.interrupted()) throw new InterruptedException
```

Comment:
This is the magic to make it work in most of the cases :smile:. Have you benchmarked it?

Reply:
No- dont do this - check can be expensive

Reply:
Removed for now. We could add it back into, say, `typedDefDef` to check far less frequently.

Reply:
It's not yet removed (it seems).

Reply:
Removed, we'll go back to for the end-of-compilation unit check.